src/libs/utils.js

```diff
@@ -369,6 +369,29 @@ export const truncateInteger = integer => {
   return `${Math.floor(integer / 1000)}k`
 }
 
+/**
+ * Polls using a given function until the pollUntil function returns true.
+ * @param {() => { result: A, shouldContinue: boolean }} pollFn - The function to poll using
+ * @param {number} pollTime - How much time there should be in ms between calls of the pollFn
+ * @param {boolean} leading - Whether the function should wait {pollTime} ms before running for the first time
+
+ * @returns {A} - The response of pollFn
+ */
+export const poll = async (pollFn, pollTime, leading = true) => {
+  do {
+    !leading && await delay(pollTime)
```

Comment:
Is this correct? Doesn't this call `await` when `leading` is `false`? If so then it waits before the first call, which is the opposite of how `leading` is described above.

(Also, isn't `leading ||` equivalent to `!leading &&`? Avoiding the `!` would make it easier to read IMO.)

Reply:
That's correct! Will fix!